src/Project/Project.router.ts

```diff
@@ -290,6 +300,62 @@ export class ProjectRouter extends Router {
     return true
   }
 
+  async getContents(req: Request, res: Response) {
+    const projectId = server.extractFromReq(req, 'id')
+    const content = server.extractFromReq(req, 'content')
+
+    if (content === INDEX_HASH) {
+      return res.send(new TextDecoder().decode(scenePreviewMain))
+    }
+
+    // when content is preview, return entity object
+    if (content === PREVIEW_HASH) {
+      try {
+        const { scene, project } = await getProjectManifest(projectId)
+        if (scene.sdk6) {
+          return res
+            .status(STATUS_CODES.badRequest)
+            .send(
+              server.sendError(
+                { projectId: project.id },
+                "Can't preview projects with sdk6 scene"
+              )
+            )
+        }
+
+        const entity = await new SDK7Scene(scene.sdk7).getEntity(project)
+
+        // Add composite file
+        entity.content = [
+          { file: "bin/index.js", hash: INDEX_HASH },
+          { file: "main.crdt", hash: CRDT_HASH },
+          ...entity.content.filter(({ file }) => file !== 'main.crdt'),
```

Comment:
is this necessary now that the `main.crdt` is stored separately?

Reply:
I added it because the scene i was testing had it in the mappings and forgot to remove it. But no, its not necessary